### ISSUE TITLE
Assertion error while using triplet_semihard_loss function

### ISSUE THREAD

**Issue by pavankoushik22** [tensorflow/tensorflow#19417]
Though my label dimensions are of the form (?,1) and embeddings dimensions (?, 128)
the loss function throwing out assertion error.

[  lshape = array_ops.shape(labels)](https://github.com/tensorflow/tensorflow/blob/4bd6243e45f7e4c9757e32e499495ac1ec28f95f/tensorflow/contrib/losses/python/metric_learning/metric_loss_ops.py#L178)

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by pavankoushik22** [comment]
Yes, i wrote a custom code.
Basically im trying to use `tf.contrib.losses.metric_learning.triplet_semihard_loss` function to calculate loss.
it takes input as labels, embeddings so when i tried to use it with Y(?,1) and Z3(?,128).
it is throwing assertion error though those are valid inputs and the exact point where im getting assertion error is [here](https://github.com/tensorflow/tensorflow/blob/4bd6243e45f7e4c9757e32e499495ac1ec28f95f/tensorflow/contrib/losses/python/metric_learning/metric_loss_ops.py#L178)


remaining fields are irrelevant so consider them as N/A

**Comment by skye** [comment]
I apologize, but I am having a hard time understanding what the problem is, where the problem is, and what version it affects. Please resubmit and pay attention to the issue template (https://github.com/tensorflow/tensorflow/issues/new). Please provide all the information it asks. Thank you.

**Comment by temptitle** [comment]
Problem still exists 

    178   lshape = array_ops.shape(labels)
--> 179   assert lshape.shape == 1
    180   labels = array_ops.reshape(labels, [lshape[0], 1])
    181 

AssertionError:
